pkg/openshift/certgen/untar.go

```diff
@@ -0,0 +1,66 @@
+package certgen
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+// UntarMasterFiles tries to untar the given data into the current
+// working directory of the running process otherwise in parentDir
+// if parentDir is provided. data is expected to be the OpenShift
+// master config bundle. Only /etc/origin/master files are untarred,
+// similar to the content that is generated in the artifacts directory
+// for Kubernetes deployments.
+func UntarMasterFiles(parentDir string, data []byte) error {
+	gr, err := gzip.NewReader(bytes.NewReader(data))
+	if err != nil {
+		return err
+	}
+	defer gr.Close()
+
+	tr := tar.NewReader(gr)
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			// end of tar archive
+			break
+		}
+		if err != nil {
+			return err
+		}
+		// We will untar only master configuration files.
+		if hdr.Typeflag != tar.TypeReg {
+			continue
+		}
+		if filepath.Dir(hdr.Name) != "etc/origin/master" {
+			continue
+		}
+		// our generated bindata hold the parent directory as part of the
+		// filename so we need to hack around that.
+		path := filepath.Join(parentDir, filepath.Base(hdr.Name))
+		ow, err := overwrite(path)
+		if err != nil {
+			return err
+		}
+		defer ow.Close()
+		if _, err := io.Copy(ow, tr); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func overwrite(mpath string) (*os.File, error) {
```

Comment:
> is there any benefit here to making sure the directory is clean vs overwriting? Just thinking that if there is ever a change in what is in the tar then we'd possibly have left over files if we just overwrite instead of blasting the dir and making everything new.

It seems that `acs-engine` will happily generate artifacts over a pre-existing directory so I wanted to stay consistent. Maybe a separate issue to discuss?